unified-codes/libs/data/src/lib/types/Entity.ts

```diff
@@ -64,10 +78,17 @@ export class Entity implements IEntity {
   }
 
   getProperty(type: string): Property {
-    const [property] = this.properties?.filter((property: IProperty) => property.type === type) ?? [];
+    const [property] =
+      this.properties?.filter((property: IProperty) => property.type === type) ?? [];
     return property;
   }
 
+  getParentProperty(type: string): Property {
+    const properties = this.getParentProperties();
+    const [property] = properties?.filter((property: IProperty) => property.type === type) ?? [];
+    return property ? new Property(property) : undefined;
```

Comment:
Something is showing up for me on this line: `TS2322: Type 'Property | undefined' is not assignable to type 'Property'.`